docs/DeveloperGuide.md

```diff
@@ -122,6 +122,7 @@ The `Model`,
 The `Storage` component,
 * can save `UserPref` objects in json format and read it back.
 * can save the address book data in json format and read it back.
+* depends on the Model component.
```

Comment:
The dependency itself should be visible in the architecture diagram (after #102 is merged). This PR can add further value by explaining why that dependency exists.

```suggestion
* depends on some classes in the `Model` component (because the `Storage` component's job it to save/retrieve objects belongs to the `Model`)
```

Reply:
@damithc I've amended this section accordingly, also I have provided a small suggestion in a comment above for the DG.